File: Benchmarks/NegatorComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class NegatorComparison {

		readonly Pattern Negator = !(Pattern)"Hello";

		readonly Regex Regex = new Regex("^[^H][^e][^l][^l][^o]");

		[Params("World", "Hello")]
		public String Source { get; set; }

		[Benchmark(Baseline = true)]
		public Result Stringier() => Negator.Consume(Source);

		[Benchmark]
		public Match MSRegex() => Regex.Match(Source);

	}
}

File: Benchmarks.FSharp/NegatorComparison.fs
namespace Benchmarks

open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type NegatorComparison() =
    let pattern = -"Hello"
    let parsec = attempt (noneOf "H" .>>. noneOf "e" .>>. noneOf "l" .>>. noneOf "l" .>>. noneOf "o")

    [<Params("World", "Hello")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark>]
    member this.ParsecRun() = run parsec this.Source